irobotclient/configuration_handler.py

```diff
@@ -0,0 +1,159 @@
+                                     description="Command line interface for iRobot HTTP requests",
+                                     usage="irobot-client [options] INPUT_FILE OUTPUT_FILE")
+    parser.add_argument("input_file", help="path and name of input file")
+    parser.add_argument("output_dir", help="path of output directory")
+    parser.add_argument("-u", "--url", help="URL scheme, domain and port for irobot. EXAMPLE: http://irobot:5000/")
+    parser.add_argument("-t", "--token", help="Arvados authentication token")
+    parser.add_argument("-f", "--force", default=False, action="store_true", help="force overwrite output file if "
+                                                                                  "it already exists")
+    parser.add_argument("--no_index", default=False, action="store_true", help="Do not download index files for"
+                                                                               "CRAM/BAM files")
+    args = parser.parse_args()
+
+    print(f'_get_command_line_args() args: {args}')  # Beth - debug
+
+    return args
+
+
+def _validate_command_line_args(args):
+    """
+    Ensures all the necessary details are set to form the requests.
+
+    :param args:
+    :return:
+    """
+    _check_input_file_argument(args)
+    _check_output_directory_argument(args)
+    _check_url_argument(args)
+    _check_authorisation_token(args)
+
+
+def _check_input_file_argument(args):
+    """
+    Strip leading slash from input argument to prevent double slashes in API request.
+
+    :param args: the command line arguments
+    """
+    if args.input_file.endswith('/') or os.path.isdir(args.input_file):
+        raise IrobotClientException(errno=errno.ECONNABORTED,
+                                    message="Cannot download entire directories at present.")
+
+    if args.input_file.startswith('/'):
+        args.input_file = args.input_file.lstrip('/')
+
+
+def _check_output_directory_argument(args):
+    """
+    Check if the output directory already exists and if it already contains files of the same name as the input file.
+
+    :return: n/a
+
+    """
+
+    try:
+        # Expand the output_dir argument so the full directory path can be used in the rest of the program.
+        args.output_dir = os.path.expanduser(args.output_dir)
+
+        # Add a trailing slash to indicate directory.
+        if not args.output_dir.endswith('/'):
+            args.output_dir = args.output_dir + '/'
+
+        # Split up the path string to obtain just the file name without extensions and full path.
+        file_name = os.path.splitext((os.path.split(args.input_file))[1])[0]
+
+        for dir_file in os.listdir(args.output_dir):
+
+            # Basic match; will return true if the directory file begins with the input_file string.
+            # TODO - Improve regex on checking whether input file already exists.
+            if re.match(os.path.splitext(dir_file)[0], file_name) and not args.force:
```

Comment:
I forgot to mention in my first reply that I have removed the extension stripping.  However, given that the initial requirements were to download data files and their associated index files, if necessary, then my initial logic seems pretty reasonable to me.  But given more information and understanding I can see the error of my ways :-)